crates/build-rs/src/input.rs

```diff
@@ -480,112 +445,159 @@ pub fn rustc_wrapper() -> Option<PathBuf> {
 /// [`build.rustc-workspace-wrapper`]: https://doc.rust-lang.org/stable/cargo/reference/config.html#buildrustc-workspace-wrapper
 #[track_caller]
 pub fn rustc_workspace_wrapper() -> Option<PathBuf> {
-    get_opt_path("RUSTC_WORKSPACE_WRAPPER")
+    var_os("RUSTC_WORKSPACE_WRAPPER").map(to_path)
 }
 
 /// The linker that Cargo has resolved to use for the current target, if specified.
 ///
 /// [`target.*.linker`]: https://doc.rust-lang.org/stable/cargo/reference/config.html#targettriplelinker
 #[track_caller]
 pub fn rustc_linker() -> Option<PathBuf> {
-    get_opt_path("RUSTC_LINKER")
+    var_os("RUSTC_LINKER").map(to_path)
 }
 
 /// Extra flags that Cargo invokes rustc with. See [`build.rustflags`].
 ///
 /// [`build.rustflags`]: https://doc.rust-lang.org/stable/cargo/reference/config.html#buildrustflags
 #[track_caller]
 pub fn cargo_encoded_rustflags() -> Vec<String> {
-    get_str("CARGO_ENCODED_RUSTFLAGS")
-        .split('\x1f')
-        .map(str::to_owned)
-        .collect()
+    to_strings(var_or_panic("CARGO_ENCODED_RUSTFLAGS"), '\x1f')
 }
 
 /// The full version of your package.
 #[track_caller]
 pub fn cargo_pkg_version() -> String {
-    get_str("CARGO_PKG_VERSION")
+    to_string(var_or_panic("CARGO_PKG_VERSION"))
 }
 
 /// The major version of your package.
 #[track_caller]
 pub fn cargo_pkg_version_major() -> u64 {
-    get_num("CARGO_PKG_VERSION_MAJOR")
+    to_parsed(var_or_panic("CARGO_PKG_VERSION_MAJOR"))
 }
 
 /// The minor version of your package.
 #[track_caller]
 pub fn cargo_pkg_version_minor() -> u64 {
-    get_num("CARGO_PKG_VERSION_MINOR")
+    to_parsed(var_or_panic("CARGO_PKG_VERSION_MINOR"))
 }
 
 /// The patch version of your package.
 #[track_caller]
 pub fn cargo_pkg_version_patch() -> u64 {
-    get_num("CARGO_PKG_VERSION_PATCH")
+    to_parsed(var_or_panic("CARGO_PKG_VERSION_PATCH"))
 }
 
 /// The pre-release version of your package.
 #[track_caller]
-pub fn cargo_pkg_version_pre() -> String {
-    get_str("CARGO_PKG_VERSION_PRE")
+pub fn cargo_pkg_version_pre() -> Option<String> {
+    to_opt(var_or_panic("CARGO_PKG_VERSION_PRE")).map(to_string)
 }
 
 /// Colon separated list of authors from the manifest of your package.
```

Comment:
Since we parse the colon separation, this description isn't accurate anymore. 

```suggestion
/// The list of authors from the manifest of your package.
```